(#1853) Hostgroups cannot be listed if filter == null

dLobatog opened the pull request:
http://theforeman.org/issues/1853

"There seems to be no easy way to configure a user such that they can see a list
of all hostgroups with out ticking all the host groups in the filter and maintaining that."

This pull requests adds a boolean field on users so that they can subscribe to new hostgroups instead of maintaining the list of hostgroups every time a new one is added. This is tedious and can be automated easily with this pull request, letting an user with view permissions be linked to every hostgroup. As an use case, we have a 'viewer' user that needs to be in all hostgroups so that we do stuff with it through the api. A comment on this ticket also mentions the same problem of having to add all hostgroups to the filter (bad maintainability).

Discussion time :)

ohadlevy commented:
I agree, we have the same problem with Orgs and hostgroups (or any other collection for that matter).

@abenari your thoughts?

ohadlevy commented:
@isratrade, this is a similar issue to what we discussed about orgs, mind having a look too? thanks!

isratrade commented:
@ohadlevy, ok, I will look into it.

ohadlevy commented:
@isratrade ping? maybe if you dont want to see whitespace changes, add ?w=1 to the url when viewing the diff

isratrade commented:
@ohadlevy, nice feature that git diff -w can be viewed by adding ?w=1
I'm fine with @eLobato's changes.

ohadlevy commented:
@eLobato do you mind rebasing and cleaning up whitespace changes?

dLobatog committed:
Users now have a new filter to subscribe to new hostgroups

Add new hostgroups on save to subscribed users

Migration to add subscribe_to_all_hostgroups to users table

I18n strings

Added tests and used a method on the controller instead of the action directly

ohadlevy commented:
[test]

ohadlevy commented:
[test]

domcleal commented:
@ohadlevy sorry, I'd configured the PR tester to ignore the old PRs (< 400) to reduce the spam.  This is now removed and it should be tested soon.

domcleal commented:
@eLobato I think it needs rebasing, GitHub is reporting it isn't mergeable.

dLobatog commented:
[test]

dLobatog commented:
[test]

isratrade commented:
@ohadlevy, I cloned and reviewed this PR and it looks fine.

ohadlevy commented:
@isratrade is that the right way forward in your opinion?

isratrade commented:
@ohadlevy, do we expect more user-specific settings in the future.  If yes, we can make a serialized attrs field rather than a boolean for each option.

dLobatog commented:
I just rebased this against develop for a smooth merge

ohadlevy commented:
merged as 5c70f9d for 1.3 thanks @eLobato!
